bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/handler/HomekitAccessoryHandler.java

```diff
@@ -0,0 +1,992 @@
+        eventedCharacteristics.clear();
+        polledCharacteristics.clear();
+
+        final Long aid = getAccessoryId();
+        if (aid == null) {
+            return;
+        }
+
+        for (Channel channel : channels.values()) {
+            final ChannelUID channelUID = channel.getUID();
+            if (isLinked(channelUID)) {
+                Long iid = 0L;
+                boolean checkChannelLinkByIID = !channelUID.equals(lightModelClientHSBTypeChannel);
+                if (checkChannelLinkByIID && channel.getProperties().get(PROPERTY_IID) instanceof String iidProperty) {
+                    try {
+                        iid = Long.parseLong(iidProperty);
+                    } catch (NumberFormatException e) {
+                        continue; // error will already have been logged elsewhere
+                    }
+                }
+
+                nestedLoops: // break marker for nested loops below
+                for (Service service : accessory.services) {
+                    for (Characteristic characteristic : service.characteristics) {
+                        if ((checkChannelLinkByIID && iid.equals(characteristic.iid))
+                                || LIGHT_MODEL_RELEVANT_TYPES.contains(characteristic.getCharacteristicType())) {
+                            Characteristic entry = new Characteristic();
+                            entry.aid = aid;
+                            entry.iid = characteristic.iid;
+                            polledCharacteristics.put(AID_IID_FORMAT.formatted(entry.aid, entry.iid), entry);
+                            if (characteristic.perms instanceof List<String> perms && perms.contains("ev")) {
+                                entry = new Characteristic();
+                                entry.aid = aid;
+                                entry.iid = characteristic.iid;
+                                eventedCharacteristics.put(AID_IID_FORMAT.formatted(entry.aid, entry.iid), entry);
+                            }
+                            if (checkChannelLinkByIID) {
+                                break nestedLoops; // unique match found; continue to next channel
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Reads the state of a specific channel by querying the accessory for the characteristic value.
+     *
+     * @param channel the channel to read
+     * @return the current state of the channel, or null if not found
+     * @throws Exception compiler requires us to handle any exception; but actually will be one of the following:
+     *             ExecutionException,
+     *             TimeoutException,
+     *             InterruptedException,
+     *             IOException,
+     *             IllegalStateException
+     */
+    private synchronized @Nullable State readChannel(Channel channel) throws Exception {
```

Comment:
It would be better do declare them explicitly.

Reply:
> It would be better do declare them explicitly

Please read the JavaDoc. These methods depend on a blocking call to a runnable that is submitted on another thread. The syntax of that is that the caller must be able to handle ANY `Exception` (the compiler cannot know what possible exceptions might occur in any specific case). Therefore the `throws Exception` is required to handle this syntax. However the JavaDoc gives the list if those exceptions which in this case are expected.

Reply:
Sorry, learned something new after reading https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/Callable.html.